src/dbnode/network/server/httpjson/handlers.go

```diff
@@ -164,8 +164,14 @@ func RegisterHandlers(mux *http.ServeMux, service interface{}, opts ServerOption
 			var in interface{}
 			if reqIn != nil {
 				in = reflect.New(reqIn.Elem()).Interface()
-				if err := json.NewDecoder(r.Body).Decode(in); err != nil {
-					writeError(w, errInvalidRequestBody)
+				decoder := json.NewDecoder(r.Body)
+				disableDisallowUnknownFields := r.Header.Get(handleroptions.DisableJSONDisallowUnknownFields)
+				if disableDisallowUnknownFields != "true" {
```

Comment:
nit: maybe `if len(disableDisallowUnknownFields) > 0`? i.e. if the header is set with any value, then allow?

Reply:
Would prefer to require explicit true, in case someone parses from config and sends "false" with each request.